amazons3/amazons3_utils.bal

```diff
@@ -19,32 +19,45 @@
 import ballerina/crypto;
 import ballerina/encoding;
 import ballerina/http;
-import ballerina/io;
 import ballerina/system;
 import ballerina/time;
 
 function generateSignature(http:Request request, string accessKeyId, string secretAccessKey, string region,
-                           string httpVerb, string requestURI, string payload) {
+                           string httpVerb, string requestURI, string payload) returns error? {
 
     string canonicalRequest = "";
     string canonicalQueryString = "";
     string stringToSign = "";
     string payloadBuilder = "";
     string authHeader = "";
-    string amzDate = "";
-    string shortDate = "";
+    string amzDateStr = "";
+    string shortDateStr = "";
     string signedHeader = "";
     string canonicalHeaders = "";
     string signedHeaders = "";
     string requestPayload = "";
     string encodedrequestURIValue = "";
     string signValue = "";
 
-    time:Time time = time:toTimeZone(time:currentTime(), "GMT");
-    amzDate = time:format(time, ISO8601_BASIC_DATE_FORMAT);
-    shortDate = time:format(time, SHORT_DATE_FORMAT);
+    time:Time|error time = time:toTimeZone(time:currentTime(), "GMT");
+    if (time is time:Time) {
+        string|error amzDate = time:format(time, ISO8601_BASIC_DATE_FORMAT);
+        string|error shortDate = time:format(time, SHORT_DATE_FORMAT);
```

Comment:
If we are returning either of these dates, we don't need to execute both of the above lines always.

Reply:
Both Data needed to generate the amazon signature header